client/common/cmdline.c

```diff
@@ -1190,11 +1191,14 @@ static BOOL freerdp_parse_username_ptr(const char* username, const char** user,
 	else if (username)
 	{
 		/* Do not break up the name for '@'; both credSSP and the
-		 * ClientInfo PDU expect 'user@corp.net' to be transmitted
-		 * as username 'user@corp.net', domain empty (not NULL!).
+		 * ClientInfo PDU expect 'AzureAD/user@corp.net' to be transmitted
+		 * as username 'AzureAD/user@corp.net',
+		 * domain empty (not NULL!).
 		 */
 		*user = username;
 		*userlen = strlen(username);
+		*domain = username;
```

Comment:
why do you set `domain` to `username`? that contradicts the description.

Reply:
if domain = NULL 
later in the auth process we will ask for it interactively 
perhaps it should be
domain = ""
domainlen = 0

Reply:
the original comment seems to intend the same thing 
"domain empty (not NULL!)."

but I didn't want to return a pointer to a char* on the stack.

Reply:
@ElliottLester then why not set it to the terminating `\0` of the input string?